02_knight_and_princess/roman_kuprii/logger.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <fstream>
+#include <iostream>
+#include <string>
+
+constexpr int COLORS_NUMBER = 5;
+
+class Graph;
+
+class Logger {
+ public:
+  static Logger& get_logger() {
+    static Logger logger;
+    return logger;
+  }
+
+  void log(std::ofstream& logfile, const std::string& text) const {
```

Comment:
Логфайл будет логично передать только один раз, в методе, например `set_logfile()`
```suggestion
  void log(const std::string& text) const {
```